operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -155,6 +167,8 @@ func extractS3ConfigSecret(s *corev1.Secret) (*storage.S3StorageConfig, error) {
 		Endpoint: string(endpoint),
 		Buckets:  string(buckets),
 		Region:   string(region),
+		RoleArn:  string(roleArn),
```

Comment:
I suggest not to pass the role_arn through the operator callstack, because the secret will be projected into environment variables for the Loki pods anyway so will this secret field become `ROLE_ARN`.

Reply:
Ahhh I see!  I hadn't thought about that! I'll improve it then